src/store/notifications/notifications.ts

```diff
@@ -16,49 +16,45 @@ export const initialState: NotificationsState = {
   pendingAlerts: []
 };
 
-interface SendAlert {
-  type: typeof SEND_ALERT;
-  payload: AlertInfo;
-}
-
-interface RemoveAlert {
-  type: typeof REMOVE_ALERT;
-  payload: AlertInfo;
-}
-
-type NotificationsAction = SendAlert | RemoveAlert;
+const prepareAlertInfo = (message: string, severity: Color = 'error') => ({
+  payload: { message, severity }
+});
 
-export const reducer = (
-  state: NotificationsState = initialState,
-  action: NotificationsAction
-): NotificationsState => {
-  switch (action.type) {
-    case SEND_ALERT:
-      return {
-        ...state,
-        pendingAlerts: [...state.pendingAlerts, action.payload]
-      };
-    case REMOVE_ALERT:
-      return {
-        ...state,
-        pendingAlerts: state.pendingAlerts.filter(
+const notificationsSlice = createSlice({
+  name: 'chess/notifications',
+  initialState,
+  reducers: {
+    sendAlert: {
+      reducer: (state, action: PayloadAction<AlertInfo>) => {
+        state.pendingAlerts.push(action.payload);
+      },
+      prepare: prepareAlertInfo
+    },
+    removeAlert: {
+      reducer: (state, action: PayloadAction<AlertInfo>) => {
+        state.pendingAlerts = state.pendingAlerts.filter(
           a => !(a.message === action.payload.message && a.severity === action.payload.severity)
-        )
-      };
-    default:
-      return state;
-  }
-};
-
-export const sendAlert = (message: string, severity: Color = 'error'): SendAlert => ({
-  type: SEND_ALERT,
-  payload: {
-    message,
-    severity
+        );
+      },
+      prepare: prepareAlertInfo
+    }
+  },
+  extraReducers: builder => {
```

Comment:
try using the other syntax here

Reply:
this syntax is recommended with TypeScript
https://redux-toolkit.js.org/api/createSlice#the-extrareducers-map-object-notation